File: Matlab/fourier(PATH)/iFourier.m
% Wrapper around "ifft" to retrieve the original time domain signal "y", given 
% Fourier coefficients "X" and the length "L" of the original time domain signal.

function [ y ] = iFourier( X, L )
    y = zeros(size(X,1), L);    %the size of the original data matrix 
    enabled = zeros(1, L);
    r = size(X,2) - 1;
    enabled(1:1+r) = 1; 
    enabled(end-r+1:end) = 1;
    
    if size(y(:,enabled==1),2) == size([X fliplr(conj(X(:,2:end)))],2)
        y(:,enabled==1) = [X fliplr(conj(X(:,2:end)))];
    end
    
    y = ifft(y, [], 2);
end

File: Matlab/fourier(PATH)/iFourier.py
def __iFourier__( self, X, L ):

    """ Wrapper around "ifft" to retrieve the original time domain signal "y", given Fourier coefficients "X" and the length "L" of the original time domain signal.
    """

    y = np.zeros( (X.shape[0], L) )  # the size of the original data matrix
    enabled = np.zeros( (1, L) )

    r = X.shape[1] - 1

    last_index = len( enabled[0,:] )
    enabled[0, 0:1+r] = 1
    enabled[0, last_index-r+1:last_index] = 1

    one = ( y[:, np.where(enabled==1)[1]] ).shape[1]
    two = np.concatenate( (X, np.fliplr(np.conj(X[:,1:None]))), axis=1 ).shape[1]
    if one == two:
        y[:, np.where(enabled==1)[1]] = np.concatenate( (X, np.fliplr(np.conj(X[:,1:None]))), axis=1 )

    y = ifft(y)

    return y